homeassistant/components/ecobee.py

```diff
@@ -0,0 +1,168 @@
+        return
+
+    # pylint: disable=unused-argument
+    def ecobee_configuration_callback(callback_data):
+        """ Actions to do when our configuration callback is called. """
+        network.request_tokens()
+        network.update()
+        setup_ecobee(hass, network, config)
+
+    _CONFIGURING['ecobee'] = configurator.request_config(
+        hass, "Ecobee", ecobee_configuration_callback,
+        description=(
+            'Please authorize this app at https://www.ecobee.com/consumer'
+            'portal/index.html with pin code: ' + network.pin),
+        description_image="/static/images/config_ecobee_thermostat.png",
+        submit_caption="I have authorized the app."
+    )
+
+
+def setup_ecobee(hass, network, config):
+    """ Setup ecobee thermostat """
+    # If ecobee has a PIN then it needs to be configured.
+    if network.pin is not None:
+        request_configuration(network, hass, config)
+        return
+
+    if 'ecobee' in _CONFIGURING:
+        configurator = get_component('configurator')
+        configurator.request_done(_CONFIGURING.pop('ecobee'))
+
+    # Ensure component is loaded
+    bootstrap.setup_component(hass, 'thermostat')
+    bootstrap.setup_component(hass, 'sensor')
+
+    hold_temp = config[DOMAIN].get(HOLD_TEMP, False)
+
+    # Fire thermostat discovery event
+    hass.bus.fire(EVENT_PLATFORM_DISCOVERED, {
+        ATTR_SERVICE: DISCOVER_THERMOSTAT,
+        ATTR_DISCOVERED: {'hold_temp': hold_temp}
+    })
+
+    # Fire sensor discovery event
+    hass.bus.fire(EVENT_PLATFORM_DISCOVERED, {
+        ATTR_SERVICE: DISCOVER_SENSORS,
+        ATTR_DISCOVERED: {}
+    })
+
+
+# pylint: disable=too-few-public-methods
+class EcobeeData(object):
+    """ Gets the latest data and update the states. """
+
+    def __init__(self, config_file):
+        from pyecobee import Ecobee
+        self.ecobee = Ecobee(config_file)
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """ Get the latest data from pyecobee. """
+        self.ecobee.update()
+        _LOGGER.info("ecobee data updated successfully.")
+
+
+def setup(hass, config):
+    """
+    Setup Ecobee.
+    Will automatically load thermostat and sensor components to support
+    devices discovered on the network.
+    """
+    # pylint: disable=global-statement, import-error
+    global NETWORK
+
+    if 'ecobee' in _CONFIGURING:
+        return
+
+    from pyecobee import config_from_file
+
+    # Create ecobee.conf if it doesn't exist
+    if not os.path.isfile(hass.config.path(ECOBEE_CONFIG_FILE)):
+        if config[DOMAIN].get(CONF_API_KEY) is None:
+            _LOGGER.error("No ecobee api_key found in config.")
+            return
+        jsonconfig = {"API_KEY": config[DOMAIN].get(CONF_API_KEY)}
+        config_from_file(hass.config.path(ECOBEE_CONFIG_FILE), jsonconfig)
+
+    NETWORK = EcobeeData(hass.config.path(ECOBEE_CONFIG_FILE))
+
+    setup_ecobee(hass, NETWORK.ecobee, config)
+
+    def stop_ecobee(event):
```

Comment:
If you're not going to use a stop event, no need to declare it or register it. And so you can remove the start listener too.